### ISSUE TITLE
Goal 7.1: Prepare midterm presentation

### ISSUE THREAD

**Issue by yumna-tree** [bschink/pocket-narrator#27]
Minimal goal & Big Vision:
- same as in first presentation probably
- one slide restating it with 1 sentence

Data: preprocessing and evaluation script on dataset

Evaluation Measures we use and how we interpret them
- grammar score note: wont exceed dataset grammar performance
- etc2

Tokenizer
- what kind
- computation of embeddings
- pos
- notably difficult things? ---> zero padding, did not expect bpe training did take so long

Present Prototypes (ngram a,b,c & transformer d,e,f,g,h,i):
- model type and architecture, params, mods, why those mods
- how they look like and how we got there
- show output samples and wandb graphs of training
- current performance and wandb graphs of eval

Surprises:
- falling right into the repetition issue
- ngram b is better than ngram c
- stumbled on the reason why we want to use linear attention and/or mamba because they are both O( n ) in contrast to O(n^2) ------> max_len in paper 512, we use 256 (us trying 512 with multi-head attention means that doubling sequence length will quadruple computation)
- [SOLVED, many methods and bugfixes to mention about this] if the issue with validation scores getting worse while training scores getting better persists its either too few epochs (just about to see improvements) or overfitting which we refuse to believe, but its not the way we calculate bcs we matched precisely between train and val, or there is some deeper issue in how we define our evaluation technically ----> need to probably reinforce early stopping

Obstacles we foresee:
- use test set for normal evaluation to see what responses we get from the models and how they score on our metrics
- question related to he upper point: should we use the test set for normal evaluation immediately after training sessions and finetune based on results frequently-> risk: we glorify the test set and overfit unconsciously [alternative would be: use the test set only in the end]
- trying to master ML witchcraft

Plan for the rest of the semester:
- next 3 goals
- what else ?
